polyinterface/polyinterface.py

```diff
@@ -35,8 +41,11 @@ def __init__(self, level):
         self.level = level
 
     def write(self, message):
-        if not re.match(r'^\s*$', message):
-            self.level(message.strip())
+        if isinstance(message, string_types):
+            if not re.match(r'^\s*$', message):
+                self.level(message.strip())
+        else:
+            self.level('ERROR: message was not a string: {}'.format(message))
```

Comment:
I've also tested with this which works, so I don't really care which is preferred.
```
    def write(self, message):
        if type(message) == type('') or type(message) == type(u'') or type(message) == type(b''):
            if not re.match(r'^\s*$', message):
                self.level(message.strip())
        else:
            self.level('ERROR: message was not a string: {}'.format(message))
```